### PR TITLE
Invalidate Discord authorization tokens on public pastes

### LINKED ISSUE #35: Automatically Invalidating Discord Tokens
[Apparently](https://github.com/PythonistaGuild/MystBin#setting-up-apps:~:text=You%20may%20set%20up%20the%20github%20bot%20token%20with%20a%20github%20personal%20access%20token%20to%20enable%20automatic%20invalidation%20of%20discord%20tokens%20(via%20posting%20them%20in%20gists).) MystBin already invalidates Discord Tokens in files, but this is not the case in practice.

I did this in a fresh file, `file.txt`, and pasted a valid discord token. After waiting for a while, nothing happened. No message from discord saying they detected my bot's token out on the internet, nothing. 

[The aforementioned file](https://mystb.in/ConspiracyForestsGrenada)

(the token in there has been invalidated already, don't even try)

### PR DESCRIPTION
**Pull request by AbstractUmbra**
## Description
Fixes #35 

This PR adds the capability to send Discord Bot tokens posted in public pastes to GitHub gists, which utilises their crawler and partnership to invalidate these tokens.

## Checklist
- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
    - [x] I have updated the changelog with a quick recap of my changes.
- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)
- [x] I have read and agree to the [Developer Certificate of Origin](https://developercertificate.org) for this contribution

### CODE REVIEW

File: views/api.py [hunk 1]
```diff
@@ -30,11 +32,69 @@
...
+            "X-GitHub-Api-Version": "2022-11-28",
+        }
+
+        for paste_id, tokens in self.__tokens_bucket.items():
+            filename = str(datetime.datetime.now(datetime.UTC)) + f"/{paste_id}-tokens.txt"
+            json_payload["files"][filename] = {"content": tokens}
+
+        await self.app.session.post("https://api.github.com/gists", headers=github_headers, json=json_payload)
+
+        self.__tokens_bucket = {}
```
Comment (LostLuma) [inline views/api.py:97]: Should swap these so tokens that are added while the request is in flight are not discarded.
```suggestion
        self.__tokens_bucket = {}

        await self.app.session.post("https://api.github.com/gists", headers=github_headers, json=json_payload)
```

File: views/api.py [hunk 2]
```diff
@@ -30,11 +32,69 @@
...
+            "Accept": "application/vnd.github+json",
+            "Authorization": f"Bearer {self._gist_token}",
+            "X-GitHub-Api-Version": "2022-11-28",
+        }
+
+        for paste_id, tokens in self.__tokens_bucket.items():
+            filename = str(datetime.datetime.now(datetime.UTC)) + f"/{paste_id}-tokens.txt"
+            json_payload["files"][filename] = {"content": tokens}
+
+        await self.app.session.post("https://api.github.com/gists", headers=github_headers, json=json_payload)
```
Comment (EvieePy) [inline views/api.py:95]: Might need to handle errors here since if this fails the entire token task will stop until the application is restarted.

Comment (EvieePy) [reply views/api.py:95]: You could also add an info log or something to say we posted to gist.

File: views/api.py [hunk 3]
```diff
@@ -30,11 +34,86 @@
...
+                    response_body = await resp.text()
+                    LOGGER.error(
+                        "Failed to create gist with token bucket with response status code %s and request body:-\n\n",
+                        resp.status,
+                        response_body,
+                    )
+                    self.__tokens_bucket.update(current_tokens)
+                    return
+        except (aiohttp.ClientError, aiohttp.ClientOSError):
+            LOGGER.info("Gist created and invalidated tokens from %s pastes.", len(current_tokens))
```
Comment (LostLuma) [inline views/api.py:116]:
```suggestion
        current_tokens = self.__tokens_bucket
        self.__tokens_bucket = {}

        for paste_id, tokens in current_tokens.items():
            filename = str(datetime.datetime.now(datetime.UTC)) + f"/{paste_id}-tokens.txt"
            json_payload["files"][filename] = {"content": tokens}

        success = False

        try:
            async with self.app.session.post(
                "https://api.github.com/gists", headers=github_headers, json=json_payload
            ) as resp:
                success = resp.ok

                if not success:
                    response_body = await resp.text()
                    LOGGER.error(
                        "Failed to create gist with token bucket with response status code %s and response body:\n\n%s",
                        resp.status,
                        response_body,
                    )
        except (aiohttp.ClientError, aiohttp.ClientOSError) as error:
            success = False
            LOGGER.error(
                "Failed to create gist with token bucket with response status code %s and request body:\n\n%s",
                error
            )

        if success:
            LOGGER.info("Gist created and invalidated tokens from %s pastes.", len(current_tokens))
        else:
            self.__tokens_bucket.update(current_tokens)
```

**Review by EvieePy — APPROVED** [review]

**Pull request merged**